MCP admin docs (#435)
mattermost/mattermost-plugin-ai: docs/admin_guide.md

- **Workflow Automation**: Use external MCP clients to automate routine tasks like posting standup updates, creating project channels, or notifying teams of important events
- **Context-Aware Assistance**: AI agents can read conversation threads to understand context before responding or taking action

### Available Tools

The MCP server provides the following tools to AI agents and external clients:

- **read_post**: Read a specific post and its thread
- **read_channel**: Retrieve recent posts from a channel
- **search_posts**: Search across Mattermost content with optional team/channel filters
- **create_post**: Create new posts or replies in channels
- **create_channel**: Create new public or private channels
- **get_channel_info**: Retrieve channel details by ID or name
- **get_team_info**: Retrieve team details by ID or name
- **search_users**: Find users by username, email, or name
- **get_channel_members**: List all members of a channel
- **get_team_members**: List all members of a team

### Deployment

![MCP Server Configuration](img/system-console-mcp.PNG)

#### For AI Agents

> The embedded MCP server provides Mattermost AI agents with direct access to Mattermost functionality:

Reviewer: ```suggestion
To set up an embedded MCP server providing Mattermost AI agents with direct access to Mattermost functionality:
```

1. Navigate to **System Console > Plugins > Agents > MCP Servers**
2. Set **Enable Embedded Server** to **True**

> 3. When enabled, all configured agents can access Mattermost tools

Reviewer: ```suggestion
3. When enabled, all configured agents can access Mattermost tools.
```

Agents will automatically use these tools when appropriate to complete user requests.

#### For External Clients

Enable external MCP clients (such as Claude web, Claude Code, or other MCP-compatible applications) to interact with your Mattermost instance.

**Requirements:**
- Mattermost Server v11.2 or later
- Valid authentication method (OAuth or Personal Access Token)

**Note:** The server uses streamable HTTP transport and does not support traditional Server-Sent Events (SSE) transport. External clients must use the streamable HTTP transport available at the `/mcp` endpoint.

**Enabling:**

1. Navigate to **System Console > Plugins > Agents > MCP Servers**

> 2. Set **Enable Mattermost MCP Server (HTTP)** to **True**

Reviewer: ```suggestion
2. Set **Enable Mattermost MCP Server (HTTP)** to **True**.
```
